src/main.cpp

```diff
@@ -1210,14 +1227,18 @@ bool CheckTransaction(const CTransaction& tx, bool fZerocoinActive, bool fReject
         }
     }
 
+    // Additional check for cold staking
+    if(tx.IsCoinStake() && !tx.HasZerocoinSpendInputs() && !CheckColdStake(tx, state))
+        return state.DoS(100, error("CheckTransaction() : invalid cold stake"), REJECT_INVALID, "bad-txns-cold-stake");
```

Comment:
Is it me or this will fail in every single coinStake tx that it's not zerocoin and cold staking?

Reply:
it fails when **all** of these conditions are satisfied
1. is a coinstake
2. is not a zpiv stake
3. it fails `CheckColdStake`.

Worth noting that "regular" stakes pass `ChecColdStake` test:
```cpp
    if (!prevOut.scriptPubKey.IsPayToColdStaking())
    	return true;
```

Reply:
After giving it a closer view, I think an even cleaner solution would be to pass the script directly from the interpreter (https://github.com/PIVX-Project/PIVX/pull/955/commits/14a3c3d0f977efdb83a999658711ae8c93b22e79).
This way we don't need additional checks in main.